packages/components/src/settings/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { Form } from '../index';
+import {
+	SettingText,
+	SettingPassword,
+	SettingCheckbox,
+	SettingSelect,
+} from './types';
+
+const typeMap = {
+	text: SettingText,
+	password: SettingPassword,
+	checkbox: SettingCheckbox,
+	select: SettingSelect,
+};
+
+export const Settings = ( {
+	fields: baseFields = [],
+	isBusy,
+	onSubmitCallback = () => {},
+	onButtonClickCallback = () => {},
+	onChangeCallback = () => {},
+	validate = () => {},
+	buttonLabel,
+} ) => {
+	// Support accepting fields in the format provided by the API (object), but transform to Array
+	const fields =
+		baseFields instanceof Array ? baseFields : Object.values( baseFields );
```

Comment:
In a README, it would be good to document the expected shape of the data.

Reply:
Yup, definitely on the agenda.